### PR TITLE
Retry connect timeouts and use ms for t/o units

### PR DESCRIPTION
**Pull request by dmuino**
Retry connection timeouts up to 3 times (hard coded for now). Use the
newer _MS curl options to specify the times in milliseconds. Currently, 
the metrics are not reflecting the attempts. This will be addressed in a 
subsequent PR as we switch to common-ipc metrics

### CODE REVIEW

File: run-build.sh [hunk 1]
```diff
@@ -27,16 +27,7 @@ else
...
-# Checks if last comand didn't output 0
-# $? checks what last command outputed
-# If output is 0 then command is succesfuly executed
-# If command fails it outputs number between 0 to 255
-if [ $? -ne 0 ]; then
-    error "Error: there are compile errors!"
-	# Terminate script and outputs 3
-    exit 3
-fi
+make -j4 || make
```
Comment (brharrington) [inline run-build.sh]: Is this a workaround because parallel builds are flakey?

**Comment by dmuino** [comment]
Yes.

File: spectator/config.h [hunk 2]
```diff
@@ -12,17 +13,17 @@ class Config {
...
         batch_size{batch_sz},
         frequency{freq},
         uri{std::move(publish_uri)} {}
   std::map<std::string, std::string> common_tags;
-  int read_timeout;     // in seconds
-  int connect_timeout;  // in seconds
+  std::chrono::milliseconds read_timeout;
+  std::chrono::milliseconds connect_timeout;
   int batch_size;
   int frequency;  // in seconds
```
Comment (troshko111) [inline spectator/config.h]: Would probably be best to have same units throughout in the ctor, a bit confusing the frequency is left in seconds.

Comment (dmuino) [reply spectator/config.h]: I'll update frequency in the next release to take milliseconds as well. Thanks for the feedback.

**Pull request merged**